lib/service/service.go

```diff
@@ -2357,6 +2367,9 @@ func (process *TeleportProcess) initProxyEndpoint(conn *Connector) error {
 		return trace.Wrap(err)
 	}
 
+	proxySSHAddr := cfg.Proxy.SSHAddr
+	proxySSHAddr.Addr = listeners.ssh.Addr().String()
```

Comment:
same as above - explain with a comment

also, this may introduce a subtle behavior change, if listener addr is different from config addr.
is that safe?

Reply:
That is the intended behavior.  These changes to handling of proxy and auth addrs allow the `tsh login` tests to work with teleport instances that bind a random port (e.g. `127.0.0.1:0`).